### ISSUE TITLE
New rule: detect GNU-only `tee --append` in shell scripts

### ISSUE THREAD

**Issue by TWiStErRob** [TWiStErRob/net.twisterrob.ghlint#757]
### Problem

`tee --append` is a GNU coreutils long option. macOS `tee` (BSD) only supports the short flag `-a`. When GitHub Actions workflows or composite actions use `tee --append` and run on macOS runners, they fail:

```
tee: illegal option -- -
usage: tee [-ai] [file ...]
jq: error: writing output failed: Broken pipe
##[error]Process completed with exit code 1.
```

The `jq` broken pipe is a cascading failure — `tee` dies first, breaking the pipe.

### Suggestion

Add a rule that flags `tee --append` in shell `run:` scripts and suggests `tee -a` instead.

`-a` is POSIX-compliant and works on all GitHub Actions runner OSes:
- **Linux** — GNU `tee` supports both `-a` and `--append`
- **macOS** — BSD `tee` only supports `-a`
- **Windows** — Git Bash (MSYS2) ships GNU `tee`, supports both

### Related existing rules

`EnvironmentFileOverwriteRule` and `SafeEnvironmentFileRedirectRule` only match `>>` redirect patterns via regex. They do not cover `| tee -a "\${GITHUB_OUTPUT}"` patterns at all, even though this is a common alternative used when you want to both append to the environment file and echo the value to the log for debugging.

Possible approaches:
1. **Standalone rule** (e.g. `GnuOnlyToolOptions`) flagging `--append` → `-a` on `tee`
2. **Extend existing environment file rules** to also recognize `| tee ... "\${GITHUB_OUTPUT}"` as an append pattern (covering both overwrite detection and safe quoting)
3. **Both** — a general portability rule plus specific environment file coverage

### Additional context

This could be generalized beyond `tee` to other common GNU-vs-BSD differences in shell scripts (e.g. `sed -i` vs `sed -i ''`, `grep -P` etc.), but `tee --append` is the most commonly hit case in GitHub Actions.

Example code:
```yml
           {
             echo "actions<<EOF"
             jq \                                          # ← jq writes to stdout
               --null-input \
               --arg workflow_url "$WORKFLOW_URL" \
               '[
                   { title: "View Workflow", url: $workflow_url }
               ]'
             echo "EOF"
           } | tee --append "${GITHUB_OUTPUT}"             # ← tee dies here, breaks pipe
```
